developer-notes.md

```diff
@@ -0,0 +1,135 @@
+# Developer Notes
+
+This guide contains some tricks and "do"s and "don't"s for new developer.
+In particular, it highlights some pitfalls one can easily trap into.
+
+# TLTR for the Impatient
+
+ 1. If you create a new Eloquent model, use the trait
+    `\App\Models\Extensions\UTCBasedTimes`.
+ 2. If you write a database migration and create a new table, do not use
+    the convenient method `\Illuminate\Database\Schema\Blueprint#timestamps`
+    in order to create the columns `created_at` and `updated_at`.
+    Instead create them manually like this
+    
+        $table->dateTime('created_at', 0)->nullable(false);
+        $table->dateTime('updated_at', 0)->nullable(false);
+    
+ 3. If you write a database migration and need a column which stores a
+    date and time, only use `\Illuminate\Database\Schema\Blueprint#dateTime`.
+ 4. Do not use the methods `\Illuminate\Database\Schema\Blueprint#timestamp`
+    nor `datetime_tz` to create a column.
+    Depending on the database back-end they are mapped to SQL types which
+    show different functional behaviour and thus are not back-end
+    agnostic.
+    Basically, `timestamp` and `datetime_tz` should be considered to be
+    buggy.
+    
+# Information on Date/Time Attributes and How Date/Time Values are Handled by the Lychee Application
+
+## Summary
+
+All date/time values are stored at the DB back-end relative to UTC without
+explicit timezone information.
+for communication between the PHP application and the DB back-end all
+date/time values are converted to SQL strings relative to UTC without
+explicit timezone indication in the string.
+For DBMS which support to set an explicit timezone for the connection
+between the DBMS server and the DBMS client of the application
+(e.g. MySQL and PostgreSQL), the timezone of the connection is set to "UTC".
+This setting resides in `./config/database.php` and should not be changed.
+The default timezone of the application as set in `./config/app.php` or
+`./.env` resp. is independent of the DBMS timezone and may be configured.
+At the level of the PHP application all date/time values are `Carbon` objects.
+Conversion of the timezone of a `Carbon` instance and UTC happens at the
+application layer during hydration/dehydration from/to the DBMS
+(that's where `\App\Models\PatchedBaseModel` comes into play).
```

Comment:
You're are right. I had to switch from a base class to a trait, because I needed that for the `User` model, too. I forgot to update the documentation. It's fixed now.